### DISCUSSION TITLE
Cannot return null for non-nullable field when throwing error

### DISCUSSION THREAD

**Discussion by erikmellum** [MichalLytek/type-graphql#1568] [Q&A]
**Describe the issue**
When I throw an error in a mutation, the custom error is being discarded in favor of "cannot return null for non-nullable field". I suspect I am doing something wrong, but I do not see in the documentation what the problem is. I am hoping that some guidance can be provided, with hopes that this will also help someone else in the future.

```javascript
@Resolver(() => TimeOffRequest)
export class TimeOffRequestCustomResolver implements ResolverInterface<TimeOffRequest> {
  @Authorized(agentEdit, supervisorEdit, adminEdit)
  @Mutation(() => TimeOffRequest)
  async createTimeOffRequest(
    @Ctx() { timeOffRequestService, request }: Context,
    @Arg('data', () => CreateTimeOffRequestInput) data: CreateTimeOffRequestInput
  ): Promise<TimeOffRequest> {
    const { accountId, userId } = getAccountIdAndUserIdFromToken(request)
    throw new GraphQLError('Invalid argument value', {
      extensions: {
        code: 'BAD_USER_INPUT',
      },
    })
}
```

When I run this, the graphql error formatter is seeing: `'Cannot return null for non-nullable field Mutation.createTimeOffRequest.'`

![image](https://github.com/MichalLytek/type-graphql/assets/6475033/d1b82d04-f250-4872-969a-783dc818fd41)

**Comment by p0thi** [comment]
@erikmellum did you find a solution for this?
I am having pretty much the same issue. After I intentionally threw a GraphQLError inside a resolver, I still get an INTERNAL_SERVER_ERROR saying, that a field on the output of the resolver is not nullable. But because I throw the error, of course I don't return anything inside the resolver. So that doesn't make sense.

I am using

- `type-graphql` with version `2.0.0-rc.1` and also tried `2.0.0-beta.3`
- `@apollo/server` version `4.9.5`

@MichalLytek do you have an idea here?

**Comment by MichalLytek** [comment]
Does it happen only with `throw new GraphQLError()` or with regular Error too?

I'm not sure if we should throw `GraphQLError` directly 🤔

**Comment by erikmellum** [comment]
@MichalLytek this happens with regular Error as well.

@p0thi no I haven't solved this problem yet. It is very important but other priorities have limited my time on this.

**Comment by MichalLytek** [comment]
Please create a minimal reproducible code example repository then.
